packages/core/android/src/test/kotlin/com/datadog/reactnative/DdSdkBridgeExtTest.kt

```diff
@@ -0,0 +1,361 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.reactnative
+
+import com.datadog.tools.unit.GenericAssert.Companion.assertThat
+import com.datadog.tools.unit.keys
+import com.datadog.tools.unit.toReadableArray
+import com.datadog.tools.unit.toReadableMap
+import com.facebook.react.bridge.JavaOnlyArray
+import com.facebook.react.bridge.JavaOnlyMap
+import com.facebook.react.bridge.WritableArray
+import com.facebook.react.bridge.WritableMap
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertDoesNotThrow
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+
+@Extensions(ExtendWith(MockitoExtension::class))
+@MockitoSettings(strictness = Strictness.LENIENT)
+internal class DdSdkBridgeExtTest {
+    // Default providers for toWritableArray and toWritableMap
+    private val createWritableMap = { JavaOnlyMap() }
+    private val createWritableArray = { JavaOnlyArray() }
+
+    @Test
+    fun `M call toWritableArray W all types of supported values`() {
+        val array = getTestArray()
+        val nestedTestMap = getTestMap()
+        val nestedTestArray = getTestArray()
+        array.add(nestedTestMap) // Nested map
+        array.add(nestedTestArray) // Nested array
+
+        val writableArray = array.toWritableArray(createWritableMap, createWritableArray)
+
+        assertDoesNotThrow {
+            assertThat(writableArray.size()).isEqualTo(array.size)
+            testWritableArray(writableArray)
+
+            // Nested map
+            val nestedMap = writableArray.getMap(array.size - 2)
+            assertThat(nestedMap).isInstanceOf(WritableMap::class.java)
+            assertThat(nestedMap.keys().size).isEqualTo(nestedTestMap.keys.size)
```

Comment:
you will have more descriptive message. same applies to similar assertions in other tests below.

```suggestion
            assertThat(nestedMap.keys()).hasSize(nestedTestMap.keys.size)
```